precompiles/randomness/src/lib.rs

```diff
@@ -384,38 +400,63 @@ where
 	) -> EvmResult {
 		let request_id = request_id.converted();
 
+		// Since we cannot compute `prepare_and_finish_fulfillment_cost` now (we don't
+		// know the number of words), we compute the cost for the maximum allowed number of
+		// words.
+		let max_prepare_and_finish_fulfillment_cost =
+			prepare_and_finish_fulfillment_gas_cost::<Runtime>(
+				<Runtime as pallet_randomness::Config>::MaxRandomWords::get(),
+			);
+
+		if handle.remaining_gas() < max_prepare_and_finish_fulfillment_cost {
+			return Err(revert(alloc::format!(
+				"provided gas must be at least {max_prepare_and_finish_fulfillment_cost}"
+			)));
+		}
```

Comment:
So now the fulfill_request caller must overestimate the gas_limit just in case request.num_words = MaxWords:::get()? Will the UI automatically do this or do we need to document is it somewhere

Reply:
Yes, it must overestimate because it can be this value.

If it is under it will revert, thus the gas estimator will increase the gas limit automatically.